build.nix

```diff
@@ -183,6 +200,11 @@ with rec
             mkdir -p $out
             cp -r target $out
 
+            ${lib.optionalString doDoc ''
+            mkdir -p $doc
+            cp -r target_doc/doc/* $doc
```

Comment:
ah, looks like all the doc ends up in `target/doc`. Might as well use a single target dir then. What do you think?

Reply:
I used a dir separate from `target` since the line `cp -r target $out` above would copy the whole `target` to `$out` and I would like to have the docs only the separate output `$doc`. 

Maybe we can change `cp -r target $out` to only copy the stuff that you actually need in `$out`.